### DISCUSSION TITLE
k8s java configurations

### DISCUSSION THREAD

**Discussion by Lory1990** [mia-platform/community#263] [Show and tell]
Hi Mia Platform!
we have been using Mia Platform and k8s for more than a year now and we want to share the Java configurations we are using 

**Quarkus**
| |  Request | Limit | Amout of performance increase in case MS does not start |
|---|---- | ---- | --- |
| CPU | 25Mi | 150Mi |25Mi  |
| Memory | 50m | 200m |50m|

**Spring**

| |  Request | Limit | Amout of performance increase in case MS does not start |
|---|---- | ---- | --- |
| CPU | 50Mi | 150Mi |25Mi  |
| Memory | 250m | 450m |50m|

**Comment by fredmaggiowski** [comment]
Hey @Lory1990 thank you for sharing your experience, we really appreciate it!

I personally haven't got much experience with Quarkus, however I happened to use Spring in some projects and noticed that with low CPU limits (such as 150m) sometimes the service couldn't start due to boot operations being CPU intensive making it slow (due to throttling) and fail to reach the liveness probe threshold in time; how did you solve this issue? Do have a particular tuning on probes you could share as well?

Thank you!

**Comment by Lory1990** [comment]
Hi @fredmaggiowski,
we had the same issues, every microservice is different and needs a tailor made configuration.

Anyway the new versions of java and Spring boot consumes less CPU than the older versions.

In any case the final solution for java on k8s is Quarkus

**Reply by fredmaggiowski** [reply]
> we had the same issues, every microservice is different and needs a tailor made configuration.

Yes, I agree; I was wondering whether you found some magic numbers that work most of the time 😉 

> Anyway the new versions of java and Spring boot consumes less CPU than the older versions.

🎉 

> In any case the final solution for java on k8s is Quarkus

That's interesting; I think some team in Mia-Platform is using Quarkus and may provide their own experience on this topic!
